src/d3a/d3a_core/sim_results/file_export_endpoints.py

```diff
@@ -15,92 +15,59 @@
-            for child in area.children:
-                self._populate_area_children_data(child)
-        self.update_plot_stats(area)
+class PastDataEnum(Enum):
+    """Enum for deciding which markets to access."""
+    SPOT_MARKET = 1
+    BALANCING_MARKET = 2
+    SETTLEMENT_MARKET = 3
 
-    @staticmethod
-    def generate_market_export_data(area, is_balancing_market):
-        return ExportBalancingData(area) if is_balancing_market else ExportData.create(area)
-
-    def _get_stats_from_market_data(self, out_dict, area, balancing):
-        data = self.generate_market_export_data(area, balancing)
-        if area.slug not in out_dict:
-            out_dict[area.slug] = dict((key, []) for key in data.labels())
-        for row in data.rows():
-            for ii, label in enumerate(data.labels()):
-                out_dict[area.slug][label].append(row[ii])
 
-    def _populate_plots_stats_for_supply_demand_curve(self, area):
-        if ConstSettings.IAASettings.MARKET_TYPE == 3:
-            if len(area.past_markets) == 0:
-                return
-            market = area.past_markets[-1]
-            if area.slug not in self.cumulative_offers:
-                self.cumulative_offers[area.slug] = {}
-                self.cumulative_bids[area.slug] = {}
-                self.clearing[area.slug] = {}
-            if market.time_slot not in self.cumulative_offers[area.slug]:
-                self.cumulative_offers[area.slug][market.time_slot] = {}
-                self.cumulative_bids[area.slug][market.time_slot] = {}
-                self.clearing[area.slug][market.time_slot] = {}
-            self.cumulative_offers[area.slug][market.time_slot] = market.state.cumulative_offers
-            self.cumulative_bids[area.slug][market.time_slot] = market.state.cumulative_bids
-            self.clearing[area.slug][market.time_slot] = \
-                bid_offer_matcher.matcher.match_algorithm.state.clearing
+past_market_type_file_suffix_dict = {
+    PastDataEnum.SPOT_MARKET: "",
+    PastDataEnum.BALANCING_MARKET: "-balancing",
+    PastDataEnum.SETTLEMENT_MARKET: "-settlement",
+}
 
-    def update_plot_stats(self, area):
-        self._get_stats_from_market_data(self.plot_stats, area, False)
-        if ConstSettings.BalancingSettings.ENABLE_BALANCING_MARKET:
-            self._get_stats_from_market_data(self.plot_balancing_stats, area, True)
-        if ConstSettings.GeneralSettings.EXPORT_SUPPLY_DEMAND_PLOTS:
-            self._populate_plots_stats_for_supply_demand_curve(area)
 
+class ExportDataBase(ABC):
 
-class ExportData:
-    def __init__(self, area):
-        self.area = area
+    @abstractmethod
+    def labels(self) -> List:
+        """Labels for csv headers."""
 
-    @staticmethod
-    def create(area):
-        return ExportUpperLevelData(area) if len(area.children) > 0 else ExportLeafData(area)
+    @abstractmethod
+    def rows(self) -> List:
+        """Return rows containing the offer, bids, trades data."""
```

Comment:
You're defining these as abstract methods, but you're implementing them as abstract properties. Pylint also detects this and is giving you a warning:
> pylint: warning
> W0236 - Method 'labels' was expected to be 'method', found it instead as 'property' (invalid-overridden-method)
So I would recommend to either implement these as methods everywhere, os to implement them as abstract properties in the ABC class:
```python
@property
@abstractmethod
def labels(self) -> List:
    # ...